server/executor/pollingprofile/polling_profile_resource.go

```diff
@@ -0,0 +1,370 @@
+package pollingprofile
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/kubeshop/tracetest/server/id"
+	"github.com/kubeshop/tracetest/server/resourcemanager"
+)
+
+type Strategy string
+
+const (
+	Periodic Strategy = "periodic"
+)
+
+const ResourceName = "PollingProfile"
+
+var Operations = []resourcemanager.Operation{
+	resourcemanager.OperationCreate,
+	resourcemanager.OperationDelete,
+	resourcemanager.OperationGet,
+	resourcemanager.OperationList,
+	resourcemanager.OperationUpdate,
+}
+
+type PollingProfile struct {
+	ID       id.ID                  `mapstructure:"id"`
+	Name     string                 `mapstructure:"name"`
+	Default  bool                   `mapstructure:"default"`
+	Strategy Strategy               `mapstructure:"strategy"`
+	Periodic *PeriodicPollingConfig `mapstructure:"periodic"`
+}
+
+type PeriodicPollingConfig struct {
+	RetryDelay string `mapstructure:"retryDelay"`
+	Timeout    string `mapstructure:"timeout"`
+}
+
+func (pp PollingProfile) HasID() bool {
+	return pp.ID.String() != ""
+}
+
+func (pp PollingProfile) Validate() error {
+	if pp.Strategy == Periodic && pp.Periodic == nil {
+		return fmt.Errorf("missing periodic polling profile configuration")
+	}
+
+	return nil
+}
+
+func NewRepository(db *sql.DB) *Repository {
+	return &Repository{db}
+}
+
+type Repository struct {
+	db *sql.DB
+}
+
+func (r *Repository) SetID(profile PollingProfile, id id.ID) PollingProfile {
+	profile.ID = id
+	return profile
+}
+
+const insertQuery = `INSERT INTO polling_profiles (
+		"id",
+		"name",
+		"default",
+		"strategy",
+		"periodic"
+	)
+	VALUES ($1, $2, $3, $4, $5)`
+
+func (r *Repository) Create(ctx context.Context, profile PollingProfile) (PollingProfile, error) {
+	var (
+		periodicJSON []byte
+		err          error
+	)
+
+	if profile.ID == "" {
+		profile.ID = id.SlugFromString(profile.Name)
+	}
```

Comment:
(to think in future) Later we need to map this behavior to document it somewhere. (maybe on OpenAPI specs?)